pcr: add initial/catchup scan metrics

kev-cao opened the pull request:
This commit adds initial and catchup scan metrics to PCR similar to the metrics supplied in steady-state LDR. The two metrics can be found with `physical_replication.scanning_ranges` and
`physical_replication.catchup_ranges`, respectively.

Fixes: #152272

Release note: Adds initial and catchup scan metrics to PCR under `physical_replication.scanning_ranges` and
`physical_replication.catchup_ranges`.

kev-cao committed:
pcr: add initial/catchup scan metrics

This commit adds initial and catchup scan metrics to PCR similar to the
metrics supplied in steady-state LDR. The two metrics can be found with
`physical_replication.scanning_ranges` and
`physical_replication.catchup_ranges`, respectively.

Fixes: #152272

Release note: Adds initial and catchup scan metrics to PCR under
`physical_replication.scanning_ranges` and
`physical_replication.catchup_ranges`.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/153893)
<!-- Reviewable:end -->

kev-cao committed:
crosscluster: move aggregate range stats collector into utils

This patch moves the aggregate range stats collector for LDR into the
replication utils package so that it can be used in PCR.

Epic: None

Release note: None

kev-cao commented on pkg/crosscluster/physical/stream_ingestion_job_test.go:
```diff
@@ -742,3 +743,59 @@ func TestPhysicalReplicationGatewayRoute(t *testing.T) {
+	clusters.WaitUntilStartTimeReached(ingestionJobID)
+	close(trackingCh)
+
+	require.Greater(t, maxScanningRanges, int64(0))
```
It takes about 2 seconds on my machine — yeah I wasn't sure if this sort of test was necessary for a metric.

That being said, it did catch bugs with my implementation, namely that race with the setting of the job status.

jeffswenson reviewed (approved):
LGTM

kev-cao commented:
TFTR!

bors r=jeffswenson
